File: server/datasets/java_source-code/Multiple-G1/renamed_37_bst.java
class Node_1 {
    int val_1;
    Node_1 lhs_1, rhs_1;
    public Node_1(int item) {
        val_1 = item;
        lhs_1 = rhs_1 = null;
    }
}
public class Tree_1 {
    public static void main(String[] args) {
        Tree_1 bst = new Tree_1();
        bst.ins_1(50);
        bst.srch_1(50);
    }
    Node_1 rt_1;
    public void ins_1(int val_1) {
        rt_1 = in_rec_1(rt_1, val_1);
    }
    Node_1 in_rec_1(Node_1 rt_1, int val_1) {
        if (rt_1 == null) return new Node_1(val_1);
        if (val_1 < rt_1.val_1) rt_1.lhs_1 = in_rec_1(rt_1.lhs_1, val_1);
        else if (val_1 > rt_1.val_1) rt_1.rhs_1 = in_rec_1(rt_1.rhs_1, val_1);
        return rt_1;
    }
    Node_1 srch_1(int val_1) {
        return sr_rec_1(rt_1, val_1);
    }
    Node_1 sr_rec_1(Node_1 rt_1, int val_1) {
        if (rt_1 == null || rt_1.val_1 == val_1) return rt_1;
        if (rt_1.val_1 < val_1) return sr_rec_1(rt_1.rhs_1, val_1);
        return sr_rec_1(rt_1.lhs_1, val_1);
    }
}

File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_g1/renamed_37_bst.py
class Node_1:
    def __init__(self, val_1):
        self.val_1 = val_1
        self.lhs_1 = None
        self.rhs_1 = None

class Tree_1:
    def __init__(self):
        self.rt_1 = None

    def ins_1(self, val_1):
        if self.rt_1 is None:
            self.rt_1 = Node_1(val_1)
        else:
            self._ins_1_rec(self.rt_1, val_1)

    def _ins_1_rec(self, nd_1, val_1):
        if val_1 < nd_1.val_1:
            if nd_1.lhs_1 is None:
                nd_1.lhs_1 = Node_1(val_1)
            else:
                self._ins_1_rec(nd_1.lhs_1, val_1)
        elif val_1 > nd_1.val_1:
            if nd_1.rhs_1 is None:
                nd_1.rhs_1 = Node_1(val_1)
            else:
                self._ins_1_rec(nd_1.rhs_1, val_1)

    def srch_1(self, val_1):
        return self._srch_1_rec(self.rt_1, val_1)

    def _srch_1_rec(self, nd_1, val_1):
        if nd_1 is None or nd_1.val_1 == val_1:
            return nd_1
        if val_1 < nd_1.val_1:
            return self._srch_1_rec(nd_1.lhs_1, val_1)
        return self._srch_1_rec(nd_1.rhs_1, val_1)
